model/service.go

```diff
@@ -44,15 +45,28 @@ type Endpoint struct {
 	// Address of the endpoint, typically an IP address
 	Address string `json:"ip_address,omitempty"`
 	// Port on the host address
-	Port int `json:"port,omitempty"`
+	Port int `json:"port"`
+	// Name of the port classifies ports for a single service
+	Name string `json:"name,omitempty"`
+	// Protocol for the port: TCP, UDP (default is TCP)
+	Protocol Protocol `json:"protocol,omitempty"`
 }
 
+// Protocol defines network protocols for ports
+type Protocol string
+
+const (
+	ProtocolHTTP Protocol = "HTTP"
```

Comment:
Yes.. And we did that based on how K8S had allowed the user to further qualify a TCP port. One example is here
http://kubernetes.io/docs/user-guide/services/#multi-port-services

Given the use of "name" to qualify the TCP port as HTTP/HTTPS etc., in Kubernetes, I assumed that your use of "name" field was for a similar intent.